### ISSUE TITLE
Reports created as extra docs in forms are not editable

### ISSUE THREAD

**Issue by dianabarsan** [medic/cht-core#7605]
**Describe the bug**
Attempting to edit a report without an content attachment keeps the user in a spinner. 

**To Reproduce**
1. Load default config. 
2. Submit a delivery form and select that the mother is deceased. 
3. (2) will generate two reports, the delivery report and a death report. 
4. Edit the death report. 

**Expected behavior**
We should not display the edit button if the form is not editable. Or at least quickly show that there;s an error instead of showing a spinner for a long time.

**Logs**
A 404 and then `Error setting selected doc`

**Screenshots**
![image](https://user-images.githubusercontent.com/35681649/165566239-f2d73aa2-a434-4213-b769-18f919e90d30.png)

**Environment**
- Instance: gamma.dev
- App: erbapp
- Version: 3.14 but been here for a long time

**Additional context**
Once https://github.com/medic/cht-core/issues/5549 lands, this behavior will change, as we won't depend on the attachment. However the report in the example is incomplete (missing "inputs" entirely), so when editing the patient won't be selected at all.
We should create another issue for that.

**Comment by Benmuiruri** [comment]
Tried to reproduce this bug [Loom video](https://www.loom.com/share/5528e4abfdc84c86aaca3bf5104b77e4) but I did not encounter the bug. #5549 might have fixed it

**Comment by dianabarsan** [comment]
Thanks for trying to reproduce. I think you're right and #5549 does change this behavior entirely. Creating reports as extra docs is not entirely seamless yet, because there could be differences in structure of forms, but that can affect any report when the form document is edited. 
I think it;s safe to close this issue as fixed. 
thanks again!
